examples/tts/aligner_g2p.py

```diff
@@ -0,0 +1,340 @@
+<\"><M AH1 L ER0><, ><M AH1 L ER0><, ><HH IY1 Z>< ><DH AH0>< ><M AE1 N><.\">
+
+Example:
+python aligner_g2p.py \
+    --model=<model_path> \
+    --manifest=<manifest_path> \
+    --out=<output_json_path> \
+    --confidence=0.02 \
+    --verbose
+"""
+
+import argparse
+import json
+import os
+import re
+
+import librosa
+import soundfile as sf
+import torch
+
+from nemo.collections.tts.models import AlignerModel
+from nemo.collections.tts.torch.helpers import general_padding
+
+
+def get_args():
+    """Retrieve arguments for disambiguation.
+    """
+    parser = argparse.ArgumentParser("G2P disambiguation using Aligner input embedding distances.")
+    # TODO(jocelynh): Make this required=False with default download from NGC once ckpt uploaded
+    parser.add_argument('--model', required=True, type=str, help="Path to Aligner model checkpoint (.nemo file).")
+    parser.add_argument(
+        '--manifest',
+        required=True,
+        type=str,
+        help="Path to data manifest. Each entry should contain the path to the audio file as well as the text in graphemes.",
+    )
+    parser.add_argument(
+        '--out', required=True, type=str, help="Path to output file where disambiguations will be written."
+    )
+    parser.add_argument(
+        '--sr',
+        required=False,
+        default=22050,
+        type=int,
+        help="Target sample rate to load the dataset. Should match what the model was trained on.",
+    )
+    parser.add_argument(
+        '--heteronyms',
+        required=False,
+        type=str,
+        default='../../scripts/tts_dataset_files/heteronyms-052722',
+        help="Heteronyms file to specify which words should be disambiguated. All others will use default pron.",
+    )
+    parser.add_argument(
+        '--confidence', required=False, type=float, default=0.0, help="Confidence threshold to keep a disambiguation."
+    )
+    parser.add_argument(
+        '--verbose',
+        action='store_true',
+        help="If set to True, logs scores for each disambiguated word in disambiguation_logs.txt.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def load_and_prepare_audio(aligner, audio_path, target_sr, device):
+    """Loads and resamples audio to target sample rate (if necessary), and preprocesses for Aligner input.
+    """
+    # Load audio and get length for preprocessing
+    audio_data, orig_sr = sf.read(audio_path)
+    if orig_sr != target_sr:
+        audio_data = librosa.core.resample(audio_data, orig_sr=orig_sr, target_sr=target_sr)
+
+    audio = torch.tensor(audio_data, dtype=torch.float, device=device).unsqueeze(0)
+    audio_len = torch.tensor(audio_data.shape[0], device=device).long().unsqueeze(0)
+
+    # Generate spectrogram
+    spec, spec_len = aligner.preprocessor(input_signal=audio, length=audio_len)
+
+    return spec, spec_len
+
+
+def get_mean_distance_for_word(l2_dists, durs, start_token, num_tokens):
```

Comment:
The same function is copied in the notebook, can we move it inside collections/tts/ to avoid duplication?

Reply:
Moved to the Aligner encoder class to match with the other Aligner utility functions.